File: cpp/organism/plant/species/Grass.cpp
//
// Created by Filip on 12/04/2025.
//

#include "Grass.h"

void Grass::reproduce() {
  Point newPosition = getNewPosition();
  if (newPosition == position)
    return; // Ensure a new position is valid

  world.killPlantsOnPosition(newPosition);
  world.queueOrganismAddition(new Grass(newPosition, world));
}

double Grass::getSpreadProbability() const {
  return spreadProbability;
}

Organism* Grass::clone() const {
    return new Grass(*this); // Tworzy kopię obiektu za pomocą konstruktora kopiującego
}

File: java/src/organism/plant/species/Grass.java
package organism.plant.species;

import constants.Constants;
import logger.EventLogger;
import organism.Organism;
import organism.plant.Plant;
import world.World;

public class Grass extends Plant {

    public Grass(int x, int y, World world) {
        super(x, y, Constants.Plant.Grass.Symbol, Constants.Plant.Grass.Strength, Constants.Plant.Initiative, world);
    }

    @Override
    protected void reproduce() {
        int[] newPosition = getNewPosition();
        if (newPosition[0] == x && newPosition[1] == y) return;
        world.addOrganism(new Grass(newPosition[0], newPosition[1], world));
        EventLogger.getInstance().log("Grass spread to (" + newPosition[0] + ", " + newPosition[1] + ")");
    }

    @Override
    protected double getSpreadProbability() {
        return Constants.Plant.Grass.SpreadProbability;
    }
}